datafusion/sql/src/statement.rs

```diff
@@ -888,7 +888,15 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 }
                 Some(v) => v,
             };
-            if !(&key.contains('.')) {
+
+            if key.to_lowercase().contains("keep_partition_by_columns") {
```

Comment:
this special case is unfortunate, but I don't have a great idea of how to make it better

Reply:
After further consideration, IMO we should restrict this handling to format options. For other configurations, users must specify the prefix. Otherwise, this list will continue to grow longer, and using those prefixes would lose their meaning (which is why all these refactors were done to have a structured configuration).

Additionally, instead of using "hive," we need a more general term. Perhaps the "execution" prefix would be a better alternative.

Reply:
I gave a try to this in https://github.com/apache/datafusion/pull/11107/commits/e91397aa4837066807f9599f9cded99649cdfc5b . Let me know if that's what you had in mind. Thanks for feedback.

Reply:
That's exactly what I had in my mind. Thank you for the collaboration. I just would like to mention two more points:
1) I guess you missed to add `execution` prefix to the example in `datafusion/sql/src/parser.rs`.
2) 
```
let keep_partition_by_columns = source_option_tuples
                    .get("execution.keep_partition_by_columns")
                    .map(|v| v.trim() == "true")
                    .unwrap_or(...
```
If the user provides anything other than "true," it is interpreted as "false." It might be wiser to give an error if the value is neither "true" nor "false."

Reply:
Good catch! I just fixed that test case and also slightly modified the handling of the value to account for what we discussed:
 - Give preference to what is explicitly provided.
 - If what is provided is invalid, through a config error. Added a test in copy.slt for this.
 - If not provided, fallback to `ExecutionOptions` value, false by default.

Reply:
This PR is a good first step towards adding settings other than format. Thanks for your effort, @hveiga. I think the PR is ready to be merged.